src/m365/spo/commands/commandset/commandset-add.ts

```diff
@@ -0,0 +1,168 @@
+      }
+    );
+  }
+
+  #initValidators(): void {
+    this.validators.push(
+      async (args: CommandArgs) => {
+        if (args.options.clientSideComponentId && !validation.isValidGuid(args.options.clientSideComponentId as string)) {
+          return `${args.options.clientSideComponentId} is not a valid GUID`;
+        }
+
+        if (SpoCommandsetAddCommand.listTypes.indexOf(args.options.listType) < 0) {
+          return `${args.options.listType} is not a valid list type. Allowed values are ${SpoCommandsetAddCommand.listTypes.join(', ')}`;
+        }
+
+        if (args.options.scope && SpoCommandsetAddCommand.scopes.indexOf(args.options.scope) < 0) {
+          return `${args.options.scope} is not a valid scope. Allowed values are ${SpoCommandsetAddCommand.scopes.join(', ')}`;
+        }
+
+        if (args.options.location && SpoCommandsetAddCommand.locations.indexOf(args.options.location) < 0) {
+          return `${args.options.location} is not a valid location. Allowed values are ${SpoCommandsetAddCommand.locations.join(', ')}`;
+        }
+
+        return validation.isValidSharePointUrl(args.options.webUrl);
+      }
+    );
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    if (this.verbose) {
+      logger.logToStderr(`Adding ListView Command Set ${args.options.clientSideComponentId} to site '${args.options.webUrl}'...`);
+    }
+
+    if (!args.options.scope) {
+      args.options.scope = 'Site';
+    }
+
+    const location: string = this.getLocation(args.options.location ? args.options.location : '');
+    const listType: string = this.getListTemplate(args.options.listType);
+
+    try {
+      const requestBody: any = {
+        Title: args.options.title,
+        Location: location,
+        ClientSideComponentId: args.options.clientSideComponentId,
+        RegistrationId: listType
+      };
+
+      if (args.options.clientSideComponentProperties) {
+        requestBody.ClientSideComponentProperties = args.options.clientSideComponentProperties;
+      }
+
+      const requestOptions: any = {
+        url: `${args.options.webUrl}/_api/${args.options.scope}/UserCustomActions`,
+        headers: {
+          accept: 'application/json;odata=nometadata'
+        },
+        data: requestBody,
+        responseType: 'json'
+      };
+      const response = await request.post<CustomAction>(requestOptions);
+
+      logger.log(response);
+    }
+    catch (err: any) {
+      this.handleRejectedODataJsonPromise(err);
+    }
+  }
+
+  private getLocation(location: string): string {
+    switch (location) {
+      case 'Both':
+        return 'ClientSideExtension.ListViewCommandSet';
+      case 'ContextMenu':
+        return 'ClientSideExtension.ListViewCommandSet.ContextMenu';
+      default:
+        return 'ClientSideExtension.ListViewCommandSet.CommandBar';
+    }
+  }
+
+  private getListTemplate(listTemplate: string): string {
+    switch (listTemplate) {
+      case 'SitePages':
```

Comment:
I was wondering for a moment that maybe we could consider reusing the `ListTemplateType` enum 🤔
but TBH there is also and mistake as `119` is actually `WikiPageLibrary` not `WebPageLibrary`. 
I wonder if `SitePages` will be understandable for the user rather than the actually template name, which is `WikiPageLibrary`

@pnp/cli-for-microsoft-365-maintainers any opinion about this?

Reply:
lets ignore this for now and leave it as is 👍